src/physics/manufactured_solution.cpp

```diff
@@ -220,6 +228,26 @@ ::value(const dealii::Point<dim,real> &point, const unsigned int istate) const
     return value;
 }
 
+template <int dim, typename real>
+inline dealii::Tensor<1,dim,real> ManufacturedSolutionZero<dim,real>
+::gradient (const dealii::Point<dim,real> &/*point*/, const unsigned int /*istate*/) const
+{
+    dealii::Tensor<1,dim,real> gradient;
+    if (dim==1) {
```

Comment:
To make it concise, this can be put in a for loop  (for(i=0;i<dim;i++) gradient[i]=0).

Reply:
Great point, I have fixed this.